scrapely/__init__.py

```diff
@@ -33,11 +35,12 @@ def train_from_htmlpage(self, htmlpage, data):
         assert data, "Cannot train with empty data"
         tm = TemplateMaker(htmlpage)
         for field, values in data.items():
-            if not hasattr(values, '__iter__'):
+            if (isinstance(values, bytes) or
+                    not hasattr(values, '__iter__')):
                 values = [values]
             for value in values:
-                if isinstance(value, str):
-                    value = value.decode(htmlpage.encoding or 'utf-8')
+                if isinstance(value, bytes):
+                    value = str_to_unicode(value, htmlpage.encoding)
```

Comment:
Actually my main issue was str_to_unicode function :) There are many ways to write such function; with .decode() it is clear what's going on. But it could be just a style preference.